arm-datalake-analytics/account/2016-11-01/swagger/account.json

```diff
@@ -1074,37 +1073,26 @@
       "description": "Azure Storage account properties information to update."
     },
     "StorageAccountInfo": {
-      "properties": {
-        "name": {
-          "type": "string",
-          "readOnly": true,
-          "description": "the account name associated with the Azure storage account."
-        },
-        "properties": {
-          "$ref": "#/definitions/StorageAccountProperties",
-          "description": "the properties associated with this storage account.",
-          "x-ms-client-flatten": true
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
         }
-      },
-      "description": "Azure Storage account information."
-    },
-    "CreateStorageAccountInfo": {
+      ],
       "required": [
-        "properties",
-        "name"
+        "properties"
       ],
       "properties": {
         "name": {
```

Comment:
Good catch I will remove it here. But see the above description. For these sub resources the name property **is** required during initial super-resource creation. For adding sub resources, these objects are not used.